main/test/ca/uwaterloo/flix/library/TestBufReader.flix

```diff
@@ -0,0 +1,187 @@
+/*
+ *  Copyright 2025 Ry Wiese
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+use Bool.∧
+
+import java.io.InputStream
+import java.io.ByteArrayInputStream
```

Comment:
I think we can do tests without Java. You can simply make a type that wraps an array and have it implement Reader, and voila you have your own ByteArrayInputStream